packages/@aws-cdk/aws-msk-alpha/lib/cluster.ts

```diff
@@ -686,37 +654,18 @@ export class Cluster extends ClusterBase {
     }
 
     let clientAuthentication;
-    if (props.clientAuthentication?.saslProps?.iam) {
-      clientAuthentication = {
-        sasl: { iam: { enabled: props.clientAuthentication.saslProps.iam } },
-      };
-      if (props.clientAuthentication?.tlsProps) {
-        clientAuthentication = {
-          sasl: { iam: { enabled: props.clientAuthentication.saslProps.iam } },
-          tls: {
-            certificateAuthorityArnList: props.clientAuthentication?.tlsProps?.certificateAuthorities?.map(
-              (ca) => ca.certificateAuthorityArn,
-            ),
-          },
-        };
-      }
-    } else if (props.clientAuthentication?.saslProps?.scram) {
-      clientAuthentication = {
-        sasl: {
-          scram: {
-            enabled: props.clientAuthentication.saslProps.scram,
-          },
-        },
-      };
-    } else if (
-      props.clientAuthentication?.tlsProps?.certificateAuthorities !== undefined
-    ) {
+    const { saslProps, tlsProps } = props.clientAuthentication ?? {};
+
+    if (props.clientAuthentication) {
```

Comment:
No need to declare them before the if
```suggestion

    if (props.clientAuthentication) {
      const { saslProps, tlsProps } = props.clientAuthentication;
```